app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -611,7 +611,7 @@ export default Controller.extend(bufferedProperty("model"), {
 
     // Post related methods
     replyToPost(post) {
-      if (this.currentUser) {
+      if (this.currentUser && this.siteSettings.enable_user_tips) {
```

Comment:
I think it'll be useful to extract the logic for this conditional into a function so that the logic is centralised instead of being duplicated in multiple places.